src/bin/pg_autoctl/cli_config.c

```diff
@@ -632,6 +632,157 @@ cli_config_set(int argc, char **argv)
 }
 
 
+static bool
+cli_keeper_config_validate_replication(KeeperConfig *config, PGSQL *pgsql)
+{
+	Keeper keeper = { 0 };
+	NodeAddress primary = { 0 };
+	bool primaryUnreachable;
+
+	if (!keeper_init(&keeper, config))
+	{
+		/* already logged */
+		return false;
+	}
+
+	/* get the primary info to populate the replication source */
+	if (!monitor_get_primary(&keeper.monitor,
+							 config->formation,
+							 keeper.state.current_group,
+							 &primary))
+	{
+		log_warn("Monitor could not be contacted for primary conninfo, skipping "
+				 "password validation");
```

Comment:
Usually our error messages start with "Failed to..." to make it explicit from the first words that it is an unexpected failure.